ovirt_imageio/client/_options.py

```diff
@@ -280,3 +281,29 @@ def __call__(self, s):
             raise ValueError(f"Size {s!r} > {self.maximum}")
 
         return value
+
+
+class Type:
+    """
+    Wrapper to expose function validators as types.
+    """
+
+    def __init__(self, name, func):
+        self.name = name
+        self.func = func
+
+    def __call__(self, value):
+        return self.func(value)
+
+    def __repr__(self):
+        return self.name
+
+
+def _validate_uuid(id):
+    try:
+        return str(uuid.UUID(id))
+    except ValueError as e:
+        raise ValueError(f"Invalid UUID value: '{id}': {str(e)}") from e
```

Comment:
You are right, I do not know why I wrote it with `str(e)`... maybe, since we wrote that so many times in tests I just replicated it here without thinking. I'll fix it.

Reply:
I have moved the except clause to the wrapper class, so that all wrapped functions will get catched and normalized their exceptions.